GVFS/GVFS.UnitTests/Mock/Git/MockGitProcess.cs

```diff
@@ -26,8 +28,18 @@ public void SetExpectedCommandResult(string command, Func<Result> result, bool m
             this.expectedCommandInfos.Add(commandInfo);
         }
 
-        protected override Result InvokeGitImpl(string command, string workingDirectory, string dotGitDirectory, bool useReadObjectHook, Action<StreamWriter> writeStdIn, Action<string> parseStdOutLine, int timeoutMs)
+        protected override Result InvokeGitImpl(
+            string command,
+            string workingDirectory,
+            string dotGitDirectory,
+            bool useReadObjectHook,
+            Action<StreamWriter> writeStdIn,
+            Action<string> parseStdOutLine,
+            int timeoutMs,
+            string gitObjectsDirectory = null)
         {
+            this.CommandsRun.Add(command);

```

Comment:
Ah, here is where we _could_ add information about the whole parameter set. Probably overkill, so ignore if you want.

Reply:
@derrickstolee not sure what you mean, can you elaborate?

Reply:
I mean you could add every input to this function as data for checking in the unit test (it could check the `gitObjectsDirectory` parameter, for instance). Again, overkill.

Reply:
Ah I see!  This is also where a GitContext would be nice as we could just have a list of them and compare.  I think we've definitely got better coverage now than we've had in the past, so as you suggest we'll punt that for now.  Good idea though!